src/internal/connector/onedrive/collection_test.go

```diff
@@ -597,3 +601,116 @@ func (suite *CollectionUnitTestSuite) TestCollectionPermissionBackupLatestModTim
+	strval := "not-important"
+
+	table := []struct {
+		name           string
+		collectionKind collectionKind
+		itemSize       int64
+		skippable      bool
+		err            error
+	}{
+		{
+			name:           "Simple item fetch no error",
+			collectionKind: CollectionKindFolder,
+			itemSize:       10,
+			err:            nil,
+		},
+		{
+			name:           "Simple item fetch error",
+			collectionKind: CollectionKindFolder,
+			itemSize:       10,
+			err:            assert.AnError,
+		},
+		{
+			name:           "malware error",
+			collectionKind: CollectionKindFolder,
+			itemSize:       10,
+			err:            clues.New("test error").Label(graph.LabelsMalware),
+			skippable:      true,
+		},
+		{
+			name:           "file not found error",
+			collectionKind: CollectionKindFolder,
+			itemSize:       10,
+			err:            clues.New("test error").Label(graph.LabelStatus(http.StatusNotFound)),
+			skippable:      true,
+		},
+		{
+			// This should create an error that stops the backup
+			name:           "small OneNote file",
+			collectionKind: CollectionKindPackage,
+			itemSize:       10,
+			err:            clues.New("test error").Label(graph.LabelStatus(http.StatusServiceUnavailable)),
+			skippable:      false,
+		},
+		{
+			name:           "big OneNote file",
+			collectionKind: CollectionKindPackage,
+			itemSize:       Size2GB,
+			err:            clues.New("test error").Label(graph.LabelStatus(http.StatusServiceUnavailable)),
+			skippable:      true,
+		},
+		{
+			// This should block backup, only big OneNote files should be a problem
+			name:           "big file",
+			collectionKind: CollectionKindFolder,
+			itemSize:       Size2GB,
+			err:            clues.New("test error").Label(graph.LabelStatus(http.StatusServiceUnavailable)),
+			skippable:      false,
+		},
+	}
+
+	for _, test := range table {
+		suite.Run(test.name, func() {
+			ctx, flush := tester.NewContext()
+			defer flush()
+
+			var (
+				t    = suite.T()
+				errs = fault.New(false)
+				item = models.NewDriveItem()
+				col  = &Collection{kind: test.collectionKind}
+			)
+
+			item.SetId(&strval)
+			item.SetName(&strval)
+			item.SetSize(&test.itemSize)
+
+			col.itemReader = func(
+				ctx context.Context,
+				hc *http.Client,
+				item models.DriveItemable,
+			) (details.ItemInfo, io.ReadCloser, error) {
+				return details.ItemInfo{}, nil, test.err
+			}
+
+			col.itemGetter = func(
+				ctx context.Context,
+				srv graph.Servicer,
+				driveID, itemID string,
+			) (models.DriveItemable, error) {
+				// We are not testing this err here
+				return item, nil
+			}
+
+			_, err := col.getDriveItemContent(ctx, item, errs)
+			if test.err == nil {
+				assert.NoError(t, err, "no error")
+			} else {
+				assert.EqualError(t, err, clues.Wrap(test.err, "downloading item").Error(), "error")
+				assert.Equal(t, test.skippable, clues.HasLabel(err, graph.LabelsSkippable), "skippable")
+			}
```

Comment:
given that all the error messages are the same, I would suggest refactoring this and doing a few things:
* use `assert.ErrorAssertionFunc` to check if there was an error or not. That would change the struct declaration to be something like `errCheck: assert.[No]Error` and the call to use it like `test.errCheck(t, err)`
* compare labels instead of error strings
* fold comparison for skippable into labels check

With the above, the code could become something like
```go
// do stuff to get results to compare against
test.errCheck(t, err)

if err == nil {
  return // nothing left to check
}

// check label(s) for error
assert.Equal(t, <map[string]struct{} of labels>, err.Labels())
```

Reply:
Ahh, that looks much cleaner.

Reply:
This looks cleaner. I've updated the code to reflect most of the items suggested here. I had to retain specifying the errors so as to add the properly labels onto it for returning from the [itemReader](https://github.com/alcionai/corso/blob/f8f35b1b4c1336da65960a4ebd616673bc8f3045/src/internal/connector/onedrive/collection_test.go#L695) function.